projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,287 @@
+            if (stopTime.getTripId().equals(trip.getTripId())) {
+
+              // From all the stops pick the stops on this trip
+              StopTime tripStops;
+              while (true) {
+                tripStops = StopTime.parseDelimitedFrom(stopTimeInputStream2);
+                if (tripStops == null) {
+                  break;
+                }
+
+                // Stops beside the chosen stop
+                if (tripStops.getTripId().equals(trip.getTripId())
+                    && tripStops.getStopId() != stopTime.getStopId()) {
+
+                  // ArrivalTime HH:mm:ss to double seconds
+                  String[] hhMMss = (tripStops.getArrivalTime().split(":"));
+                  Double arrivalTime =
+                      Double.parseDouble(hhMMss[0]) * 3600
+                          + // H
+                          Double.parseDouble(hhMMss[1]) * 60 // m
+                          + Double.parseDouble(hhMMss[2]); // s
+
+                  stopsAroundList.add(new StopsAround(tripStops, arrivalTime));
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  public void draw_map() {
+
+    // color1 to color2
+    Color c1 = new Color(255, 208, 235);
+    Color c2 = new Color(126, 0, 67);
+
+    // Normalize data
+    int maxData = 0;
+    int minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Create image
+    /** TODO: Try different size of image */
+    final BufferedImage image =
+        new BufferedImage(
+            heatMap[0].length - 1000, heatMap.length - 1000, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (heatMap[i][j] == 0) {
+          System.out.println("ERROR:" + "\n" + "heatmap matrix value should not be 0");
+        }
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r = (int) (((float) c2.getRed() - (float) c1.getRed()) * value + (float) c1.getRed());
+        int g =
+            (int) (((float) c2.getGreen() - (float) c1.getGreen()) * value + (float) c1.getGreen());
+        int b =
+            (int) (((float) c2.getBlue() - (float) c1.getBlue()) * value + (float) c1.getBlue());
+
+        Color c = new Color(r, g, b);
+        graphics2D.setPaint(c);
+        graphics2D.fillRect(i, j, 2, 2);
+      }
+    }
+
+    graphics2D.dispose();
+    // Save file
+    boolean createImage = false;
+    try {
+      createImage = ImageIO.write(image, "png", new File(imagePath.get()));
```

Comment:
It's better to have `imagePath.get()` passed to the method.

Reply:
Do you mean that i don't need "new File" there?